src/adapter/src/coord/indexes.rs

```diff
@@ -51,7 +51,7 @@ impl DataflowBuilder<'_> {
                 match self.catalog.get_entry(&id).item() {
                     // Unmaterialized view. Search its dependencies.
                     CatalogItem::View(view) => {
-                        todo.extend(view.optimized_expr.0.depends_on());
+                        todo.extend(view.optimized_expr().depends_on());
```

Comment:
It might be sufficient to do `view.expr.depends_on()` here (you might overestimate the set produced after optimization though, because some optimizations remove redundant inputs).